crates/middleware/src/main.rs

```diff
@@ -22,12 +21,14 @@ async fn main() -> Result<()> {
     // let articles_cached_left = run_ripgrep_service_and_index(config_state.clone(),needle.clone(), haystack).await;
     // println!("articles_cached_left: {:#?}", articles_cached_left.clone());
 
-    let articles_cached = search_haystacks(config_state.clone(), search_query.clone()).await?;
+    config_state.articles_cached = search_haystacks(config_state.clone(), search_query.clone()).await?;
```

Comment:
The one downside I see with making `articles_cached` a part of `ConfigState` is that we couple two separate concerns together: configuration handling and article handling. Do we always need both in all places? If the answer is no, then they shouldn't be coupled on that level. 😉 (To make that clearer, `ConfigState` could be renamed to `ConfigSync` to indicate that it is just a `Sync` version of `Config`.)

I think we should keep those two things separate.

If we need to couple them somehow, I would maybe introduce a new higher-level struct like `State { config_state, cached_articles }`
or introduce a `Cache` struct, which holds the cached articles: `Cache { articles }`.

I think this will make testing easier.

Reply:
I created a ticket https://github.com/terraphim/terraphim-ai/issues/45 so we will not forget it. I will add a comment with my thoughts there.